src/main/groovy/org/owasp/dependencycheck/gradle/tasks/AbstractAnalyze.groovy

```diff
@@ -321,6 +324,23 @@ abstract class AbstractAnalyze extends DefaultTask {
         }
     }
 
+    /**
+     * Checks whether the given project should be scanned
+     * because either scanProjects is empty or it contains the
+     * project's path.
+     */
+    def shouldBeScanned(Project project) {
+        !config.scanProjects || config.scanProjects.contains(project.path)
+    }
+
+    /**
+     * Checks whether the given project should be skipped
+     * because skipProjects contains the project's path.
+     */
+    def shouldBeSkipped(Project project) {
+        config.skipProjects.contains(project.path)
```

Comment:
On the above `shouldBeScanned` you added a `!config.scanPojects ||`...  Should that be replicated here or removed from above.  I don't think the null check is actually required as you are declared the lists as empty lists using `List<String> skipProjects = []`.  Thoughts?

Reply:
Hi @jeremylong thanks very much for your prompt reply. There are two reasons that it has been implemented in this way.  Firstly, I want to keep the logic consistent with `scanConfigurations/skipConfigurations`; it could not be replicated in `shouldBeSkipped` or removed from above otherwise `shouldBeScanned(project) && !shouldBeSkipped(project)` will be false when both `scanProjects` and `skipProjects` are empty lists. Secondly, I didn't think it was only a null check. It returns true when `config.scanProjects` is an empty list.